launcher/mainwindow_moc.cpp

```diff
@@ -130,21 +130,42 @@ MainWindow::MainWindow(QWidget * parent)
 
 void MainWindow::detectPreferredLanguage()
 {
-	auto preferredLanguages = QLocale::system().uiLanguages();
+	// Skip autodetection if language is already set to a valid VCMI language
+	std::string currentLanguage = settings["general"]["language"].String();
+
+	if (!currentLanguage.empty())
+	{
```

Comment:
This condition will always be true

Reply:
Sort of. Probably remove entire block (and attempts by installer to set game language).

At the moment I don't see clean way to pass information from installer to launcher. One option would be to use command line, but probably not a quick fix & don't see it as critical enough - system language is probably sane default in most cases.